src/client/public/scripts/components/row-element.ts

```diff
@@ -0,0 +1,85 @@
+import './row-header.js';
+
+import {html} from '../../../../../node_modules/lit-html/lib/lit-extended.js';
+import {TemplateResult} from '../../../../../node_modules/lit-html/lit-html.js';
+import {timeToString} from '../dash/utils/time-to-string.js';
+
+import {BaseElement, property} from './base-element.js';
+
+export interface StatusDisplay {
+  type: 'complete'|'actionable'|'activity'|'passive';
+  text: string;
+}
+
+export interface RowData {
+  id: string;
+  status: StatusDisplay;
+  createdAt: number;
+  author: string;
+  avatarUrl: string;
+  url: string;
+  title: string;
+  owner: string;
+  repo: string;
+  hasNewActivity: boolean;
+}
+
+export interface RowEvent {
+  text: string|TemplateResult;
+  time?: number;
+  url?: string;
+  classes?: string[];
+}
+
+export class RowElement extends BaseElement {
+  @property() data: RowData|null = null;
+  @property() extraHeaderData: TemplateResult[] = [];
+  @property() events: RowEvent[] = [];
+
+  _renderEvent(event: RowEvent) {
+    const timeTemplate = (time: number) =>
+        html`<time class="dashboard-row-event__time" datetime="${
+            new Date(time).toISOString()}">${timeToString(time)}</time>`;
```

Comment:
Is there a need for lit-html here vs just passing the template literal?

Reply:
Yes, it must be a lit tagged template literal, otherwise we will end up with it as a string in the DOM instead of these being converted to elements.